features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/TreeViewTraversal.kt

```diff
@@ -104,13 +109,29 @@ internal class TreeViewTraversal(
         return mappers.firstOrNull { it.supportsView(view) }?.getUnsafeMapper()
     }
 
+    private fun getSamplingRateForMapper(mapperName: String): Float {
+        return when (mapperName) {
+            ViewWireframeMapper::class.simpleName -> MethodCallSamplingRate.RARE.rate
+            TextViewMapper::class.simpleName -> MethodCallSamplingRate.RARE.rate
+            ImageViewMapper::class.simpleName -> MethodCallSamplingRate.RARE.rate
+            DecorViewMapper::class.simpleName -> MethodCallSamplingRate.REDUCED.rate
+            ButtonMapper::class.simpleName -> MethodCallSamplingRate.REDUCED.rate
+            else -> MethodCallSamplingRate.DEFAULT.rate
+        }
+    }
+
     data class TraversedTreeView(
         val mappedWireframes: List<MobileSegment.Wireframe>,
         val nextActionStrategy: TraversalStrategy
     )
 
-    companion object {
-        const val METHOD_CALL_SAMPLING_RATE = 0.1f
-        private const val METHOD_CALL_MAP_PREFIX: String = "Map with"
+    internal enum class MethodCallSamplingRate(val rate: Float) {
+        DEFAULT(rate = 0.1f),
+        REDUCED(rate = 0.01f),
+        RARE(rate = 0.001f)
+    }
```

Comment:
> [!NOTE]
> Woudln't it make sense to keep those name and values shared in the `MethodCalledTelemetry` class itself, so that it could be reused in other places as well?

Reply:
MethodCalledTelemetry is internal to the core and I don't want to modify the visibility for this, but I've extracted the enum to its own file